### PR TITLE
Provide system information

### LINKED ISSUE #7189: Resources endpoint does not provide mainboard and system information
[MAAS is still relying on lshw to gather mainboard information.](https://git.launchpad.net/maas/tree/src/metadataserver/builtin_scripts/hooks.py#n364) The following items are missing:

* System UUID
* System vendor
* System product
* System version
* System serial
* Mainboard vendor
* Mainboard product
* Mainboard version(BIOS/UEFI firmware version)
* Mainboard date(BIOS/UEFI firmware release date)

### PR DESCRIPTION
**Pull request by monstermunchkin**
This resolves #7189.

### CODE REVIEW

**Comment by stgraber** [comment]
Jenkins not happy

**Comment by stgraber** [comment]
@monstermunchkin 

```
Found some typos
shared/api/resource.go:20:8: "extention" is a misspelling of "extension"
```

**Comment by monstermunchkin** [comment]
Fixed the typo.

File: lxd/resources/system.go [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+	// Cache the system type
+	if systemType == "" {
+		// According to the man page, if a virtualization technology is detected, 0 is returned, a non-zero code otherwise.
+		out, err := shared.RunCommand("systemd-detect-virt")
+		if err != nil {
+			systemType = "physical"
+		} else {
+			systemType = out
+		}
```
Comment (stgraber) [inline lxd/resources/system.go:30]: So I don't think we want to show `lxc` but just `physical`, `container` or `virtual-machine`.

Should we just use `systemd-detect-virt --container` and `systemd-detect-virt --vm`, if both return non-zero, then go physical?

File: lxd/resources/system.go [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+	system.Motherboard, err = getSystemMotherboard()
+	if err != nil {
+		return nil, err
+	}
+
+	return &system, nil
+}
+
+func getSystemFirmware() (*api.ResourcesSystemFirmware, error) {
```
Comment (stgraber) [inline lxd/resources/system.go:141]: `systemGetFirmware` would be more namespaced

File: lxd/resources/system.go [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+		}
+
+		firmware.Version = strings.TrimSpace(string(content))
+	}
+
+	return &firmware, nil
+}
+
+func getSystemChassis() (*api.ResourcesSystemChassis, error) {
```
Comment (stgraber) [inline lxd/resources/system.go:183]: `systemGetChassis` would be more namespaced

File: lxd/resources/system.go [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+		}
+
+		chassis.Version = strings.TrimSpace(string(content))
+	}
+
+	return &chassis, nil
+}
+
+func getSystemMotherboard() (*api.ResourcesSystemMotherboard, error) {
```
Comment (stgraber) [inline lxd/resources/system.go:276]: `systemGetMotherboard` would be more namespaced

**Comment by stgraber** [comment]
Needs a rebase too

File: lxd/resources/system.go (hunk 1 above)
Comment (monstermunchkin) [reply lxd/resources/system.go:30]: > Should we just use systemd-detect-virt --container and systemd-detect-virt --vm, if both return non-zero, then go physical?

Yes, we should do that. I thought `systemd-detect-virt` would only output `VM` or `Container` but apparently I was wrong ;)

**Pull request merged**